File: level_2/narrative1_aggregate_weights_by_value/langs/compiler.cpp
int main() 
{
    int m, n;
    cin >> m;
    int set1[m][2];
    for (int i = 0; i < m; i++) 
    {
        cin >> set1[i][0];
        cin >> set1[i][1];
    }
    cin >> n;
    int set2[n][2];
    for (int i = 0; i < n; i++) 
    {
        cin >> set2[i][0];
        cin >> set2[i][1];
    }

    int resultSize;
    int** result = aggregate_weights_by_value(m, set1, n, set2, resultSize);

    for (int i = 0; i < resultSize; ++i)
        cout << result[i][0] << " " << result[i][1] << endl;


    return 0;
}

File: level_2/narrative1_aggregate_weights_by_value/langs/compiler.py
def main():
    m = int(input())
    set1 = []
    for i in range(m):
        key, value = map(int, input().split())
        set1.append([key, value])
    n = int(input())
    set2 = []
    for i in range(n):
        key, value = map(int, input().split())
        set2.append([key, value])

    result = aggregate_weights_by_value(m, set1, n, set2)

    for i in range(len(result)):
        print(result[i][0], result[i][1])
        
if __name__ == "__main__":
    main()